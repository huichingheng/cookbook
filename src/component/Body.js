import React, { Component } from "react";
import SideBar from "./SideBar";
import Display from "./Display";
import seedData from "../utils/seedData";
import "../style/Body.css";
import { URL, API_KEY } from "../utils/getHelper";

class Body extends Component {
  constructor() {
    super();
    this.state = {
      isHelpClicked: false,
      data: [],
      selectedRecipe: null
    };
    this.userInput = "";
  }

  searchClickHandler = async () => {
    const url = URL(API_KEY, this.userInput);
    const response = await fetch(url);

    if (response.status === 200) {
      const recipesData = await response.json();
      this.setState({ data: recipesData.recipes });
    }
    // For testing, use seedData.
    // this.setState({ data: seedData });
  };

  userInputEventListener = event => {
    this.userInput = event.target.value;
  };

  setSelectedRecipe = id => {
    const selectedRecipe = this.state.data.find(recipe => {
      if (recipe.recipe_id === id) {
        return recipe;
      }
    });
    this.setState({ selectedRecipe: selectedRecipe });
  };

  render() {
    const { data, selectedRecipe } = this.state;
    return (
      <div className="body">
        <Display className="display" selectedRecipe={selectedRecipe} />
        <SideBar
          className="side-bar"
          data={data}
          handleClick={this.setSelectedRecipe}
          handleUserInput={this.userInputEventListener}
          handleSearchClick={this.searchClickHandler}
          selectedRecipe={selectedRecipe}
        />
      </div>
    );
  }
}

export default Body;
