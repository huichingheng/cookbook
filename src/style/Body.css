.body {
  display: grid;
  grid-template-columns: 1fr 2fr;
  margin: 0px auto;
  height: calc(100vh - 121px);
  
}

@media all and (max-width: 850px) {
  .body{
    grid-template-columns: 1fr;
  }
}
