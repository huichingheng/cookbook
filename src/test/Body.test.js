import Body from "../component/Body";
import {shallowRender, isInArray} from "./helper/testHelper"

const result = shallowRender(Body)
const components = result.props.children.map(child => {
  return child.type.name;
});

describe("<Body /> Components are", () => {
  it("Display", () => {
    expect(isInArray("Display", components)).toBe(true);
  });
  it("SideBar", () => {
    expect(isInArray("SideBar", components)).toBe(true);
  });
  it("SideBar", () => {
    expect(isInArray("SideBar", components)).toBe(true);
  });

});

describe('Snapshot', () => {
  it('should match', () => {
    expect(result.props).toMatchSnapshot()
  });
});
