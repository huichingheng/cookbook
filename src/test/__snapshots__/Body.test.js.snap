// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should match 1`] = `
Object {
  "children": Array [
    <SideBar
      className="side-bar"
      data={Array []}
/>,
    <Display
      className="display"
/>,
  ],
  "className": "body",
}
`;
